app.js

```diff
@@ -0,0 +1,205 @@
+
+            const regionData = {};
+            allData.forEach(country => {
+                const region = country.region || 'Unknown';
+                if (!regionData[region]) {
+                    regionData[region] = { count: 0, population: 0 };
+                }
+                regionData[region].count += 1;
+                regionData[region].population += Number(country.population) || 0;
+            });
+
+            let resultsHTML = `
+                <div><strong>Total countries result: </strong>${totalCountries.toLocaleString()}</div>
+                <div><strong>Total Countries Population: </strong>${totalPopulation.toLocaleString()}</div>
+                <div><strong>Average Population: </strong>${avgPopulation.toLocaleString()}</div>
+                <br>
+                <table>
+                    <thead>
+                        <tr>
+                            <th>Country Name</th>
+                            <th>Total Population</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+            `;
+
+            resultsHTML += allData
+                .map(country => {
+                    const countryName = country.name?.common || 'Unknown';
+                    const population = country.population ? Number(country.population) : 0;
+
+                    return `
+                        <tr>
+                            <td>${countryName}</td>
+                            <td>${population.toLocaleString()}</td>
+                        </tr>
+                    `;
+                }).join('');
+
+            resultsHTML += `
+                    </tbody>
+                </table>
+                <br>
+                <table>
+                    <thead>
+                        <tr>
+                            <th>Region</th>
+                            <th>Number of countries</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+            `;
+
+            resultsHTML += Object.entries(regionData)
+                .map(([region, data]) => `
+                    <tr>
+                        <td>${region}</td>
+                        <td>${data.count}</td>
+                    </tr>
+                `).join('');
+
+            resultsHTML += `
+                    </tbody>
+                </table>
+            `;
+
+            statsDiv.innerHTML = resultsHTML;
+        } catch (err) {
+            showError('Error fetching all countries data: ' + err.message);
+        }
+    };
+
+    const renderSearchResults = async (searchTerm) => {
+        if (!searchTerm.trim()) {
+            showError('Please enter a country name');
+            return;
+        }
+
+        try {
+            const searchData = await getData(`https://restcountries.com/v3.1/name/${encodeURIComponent(searchTerm.trim())}`);
+
+            if (searchData.length === 0) {
```

Comment:
this is apparently not the correct way to check it:
<img width="634" height="967" alt="Screenshot 2025-08-12 at 15 50 02" src="https://github.com/user-attachments/assets/c956e01b-1e35-4b44-8d12-70ce4afe82b6" />